lib/ruby_routes/utility/path_utility.rb

```
@@ -56,10 +57,20 @@ def normalize_method(method)
 
-        path_without_query.split('/').reject(&:empty?)
+        # Optimized trimming: avoid string allocations when possible
+        start_idx = path.start_with?('/') ? 1 : 0
+        end_idx = path.end_with?('/') ? -2 : -1
+
+        if start_idx == 0 && end_idx == -1
+          path.split('/').reject(&:empty?)
+        else
+          path[start_idx..end_idx].split('/').reject(&:empty?)
+        end
       end
```

RuboCop (1.76.1)

[convention] Cyclomatic complexity for `split_path` is too high. [9/7]

(Metrics/CyclomaticComplexity)